README.md
<p align="center">
    <a href="http://www.yiiframework.com/" target="_blank">
        <img src="https://www.yiiframework.com/files/logo/yii.png" width="400" alt="Yii Framework" />
    </a>
    <h1 align="center">Yii Framework Application</h1>
    <br>
</p>

This package is [Yii Framework] application best for rapidly creating projects.

This template, contains the basic features including user login/logout and a contact page.
It includes all commonly used configurations that would allow you to focus on adding new
features to your application.

[Yii Framework]: http://www.yiiframework.com/

DIRECTORY STRUCTURE
-------------------

```
assets/             contains assets definition
commands/           contains console commands (controllers)
components/         contains web components classes mainly (user interface)
config/             contains application configurations
controllers/        contains Web controller classes
docs/               contains MySQL script initial configuration of the database.
helpers/            contains helper functions for Yii Framework 2.0.  
mail/               contains view files for e-mails
messages/           contains messages traduction in language spanish
models/             contains model classes
  forms/            contains model classes of forms
  queries/          contains common queries in the data model 
  search/           contains model classes of search mainly used in index view
runtime/            contains files generated during runtime
public/             contains the entry script for a web server  
tests/              contains various tests for the basic application
views/              contains view files for the Web application
vendor/             contains dependent 3rd-party packages
composer.json
sonar-project.properties Sonarqube configuration for sonar-scanner
```

REQUIREMENTS
------------
 

The minimum requirement by this project template that your Web server supports PHP 7.0.


INSTALLATION
------------

You can install this project template using the following command:

~~~

git clone https://github.com/appwebd/yii2-base.git

~~~

CONFIGURATION
-------------

### MySQL Database configuration

Edit the file `config/db.php` with real data, for example:

```php
return [
    '__class' => \yii\db\Connection::class,
    'dsn' => 'mysql:host=localhost;dbname=db_base',
    'username' => 'user',
    'password' => 'the-user-password',
    'charset' => 'utf8',
];
```

In the directory docs/ (root directory), you can found a MySQL script db_base.sql; With it you can re-create the database initial db_base

~~~

mysql -u root -p < docs/db_base.sql    

~~~

The initial administrator account like user and password is admin admin

**NOTES:**
- Yii won't create the database for you, this has to be done manually before you can access it.
- Check and edit the other files in the `config/` directory to customize your application as required.
(changing for example language).
- Refer to the README in the `tests` directory for information specific to basic application tests.


TESTING
-------

Tests are located in `tests` directory.


CHANGELOG
45f703c Tue Feb 11 21:38:16 2020 -0300 Added Changelog
4be37f8 Tue Feb 11 21:35:56 2020 -0300 fixed supportedLang variable.
4aad5ac Sat Feb 1 20:57:16 2020 -0300 Code cleanup, rearrange code, optimization of import
bcdae9c Sat Feb 1 20:57:05 2020 -0300 Code cleanup, rearrange code, optimization of import
47e2d9e Sat Feb 1 20:56:51 2020 -0300 Code cleanup, rearrange code, optimization of import
cdb4842 Sat Feb 1 20:56:40 2020 -0300 Code cleanup, rearrange code, optimization of import
2062357 Sat Feb 1 20:56:27 2020 -0300 Was added register any error to database.
046429a Sat Feb 1 20:55:30 2020 -0300 Was added getPostVar function
63579cc Sat Feb 1 20:55:04 2020 -0300 Was added deleteRecord function
f0b54b1 Sat Feb 1 20:54:43 2020 -0300 Code cleanup, rearrange code, optimization of import
e8f8536 Sat Feb 1 20:54:19 2020 -0300 Code cleanup, rearrange code, optimization of import
8f20543 Sat Feb 1 20:54:08 2020 -0300 Code cleanup, rearrange code, optimization of import
18934d6 Sat Feb 1 20:53:48 2020 -0300 Code cleanup, rearrange code, optimization of import
a934cf5 Sat Feb 1 20:53:23 2020 -0300 register error in database
67d0225 Sat Feb 1 20:52:41 2020 -0300 Was added a control of error database  connection
012ce30 Mon Sep 9 21:03:31 2019 -0300 Code cleanup, rearrange code, optimization of import
3c2bb8f Sat Jun 29 16:43:01 2019 -0400 was removed duplicated blocks
6219e99 Tue Jun 25 23:01:00 2019 -0400 was removed static calls
06d2039 Tue Jun 25 22:58:39 2019 -0400 was removed static calls
a64cf17 Tue Jun 25 22:58:06 2019 -0400 was removed static calls
ecc3826 Tue Jun 25 22:56:44 2019 -0400 remove blank lines
ee6e6a1 Wed Jun 19 23:51:46 2019 -0400 Code cleanup, rearrange code, optimization of import
4fc9eaa Wed Jun 19 23:05:02 2019 -0400 Code cleanup, rearrange code, optimization of import
85b2235 Wed Jun 19 23:04:29 2019 -0400 file was deleted (not used in the web application)
6bd9ff7 Wed Jun 19 22:31:40 2019 -0400 Code cleanup, rearrange code, optimization of import
1943b6a Wed Jun 19 22:31:01 2019 -0400 New column in logs table added functionCode.
c1d6838 Wed Jun 19 21:27:46 2019 -0400 Code cleanup, rearrange code, optimization of import
3e63f90 Wed Jun 19 20:28:46 2019 -0400 README.md renamed
b34d542 Wed Jun 19 20:26:26 2019 -0400 cleaning of code  with climatecode recommendations (class only 250 lines, 20 methods per class, etc.)
64fdbee Wed Jun 19 20:23:48 2019 -0400 updated gulp components
8c9ccca Wed Jun 19 20:22:29 2019 -0400 rearrange code
5076bb3 Wed Jun 19 20:21:45 2019 -0400 Updated sonarqube, packages.json, composer.json
6d77df4 Wed Jun 19 20:20:29 2019 -0400 company objects was deleted
ca28d7f Wed Jun 19 20:05:57 2019 -0400 New structure of directories
6a668c0 Thu Apr 11 16:44:01 2019 -0400 was removed some classes of file css/site.css
415eb47 Thu Apr 11 16:29:43 2019 -0400 Code cleanup, rearrange code, optimization of import
1290853 Thu Apr 11 16:18:03 2019 -0400 Code cleanup, rearrange code, optimization of import
5bea25e Thu Apr 11 16:17:28 2019 -0400 Code cleanup, rearrange code, optimization of import and was added try-catch
348ef5b Thu Apr 11 16:17:09 2019 -0400 Code cleanup, rearrange code, optimization of import
197b040 Thu Apr 11 16:16:59 2019 -0400 Code cleanup, rearrange code, optimization of import
ccb581f Thu Apr 11 16:16:40 2019 -0400 was added more constants.
979f19a Tue Mar 19 16:07:33 2019 -0300 rearrange code
1817325 Tue Mar 19 16:07:22 2019 -0300 Code cleanup, rearrange code, optimization of import
a2ff89f Tue Mar 19 16:07:03 2019 -0300 Code cleanup, rearrange code, optimization of import
5f8f783 Tue Mar 19 16:06:34 2019 -0300 Code cleanup, rearrange code, optimization of import
943c938 Tue Mar 19 16:06:15 2019 -0300 Code cleanup, rearrange code, optimization of import
16c6ad9 Tue Mar 19 16:02:41 2019 -0300 Wrong text success in constant transaction_module was removed
a70def1 Fri Mar 15 23:12:13 2019 -0300 was removed constant from /app/web/constant.php and created again in app/component/Uicomponent.
89eb93a Fri Mar 15 23:12:01 2019 -0300 was removed constant from /app/web/constant.php and created again in app/component/Uicomponent.
7f80f16 Fri Mar 15 23:10:17 2019 -0300 removed import: is never used.
109638f Fri Mar 15 23:09:58 2019 -0300 removed import: is never used.
6f64378 Fri Mar 15 23:09:38 2019 -0300 removed import: is never used.
fa2e0f4 Fri Mar 15 23:09:24 2019 -0300 removed import: is never used.
84f4247 Fri Mar 15 11:22:55 2019 -0300 Titles standarization
6ad3da7 Fri Mar 15 11:22:40 2019 -0300 was removed duplicated code.
d8ea077 Fri Mar 15 11:21:52 2019 -0300 was added a space blank in the  view title
540a8d2 Thu Mar 14 10:17:40 2019 -0300 Improvement in documentation of code
ba4f34d Thu Mar 14 10:16:52 2019 -0300 was added class CSS3 (style) of DetailView component Yii2 for custom viewer.
455138b Thu Mar 14 09:16:50 2019 -0300 Was removed all UiComponent:::warning functions
bec83f8 Thu Mar 14 09:15:38 2019 -0300 Was removed all UiComponent:::warning functions
4c6afc6 Thu Mar 14 09:15:28 2019 -0300 Was removed a duplicate block of code.
1231321 Thu Mar 14 09:15:06 2019 -0300 Was removed a duplicate block of code.
70f218a Thu Mar 14 09:14:57 2019 -0300 Was removed a duplicate block of code.
8d3c962 Thu Mar 14 09:13:47 2019 -0300 Was removed all UiComponent:::warning functions
fa003a5 Thu Mar 14 09:13:30 2019 -0300 Was removed all UiComponent:::warning functions
3604391 Thu Mar 14 09:13:11 2019 -0300 was added class CSS3 (style) of DetailView component Yii2 for custom viewer.
08065d1 Thu Mar 14 09:11:40 2019 -0300 Remove column company_id of table User
6a81d05 Thu Mar 14 09:11:10 2019 -0300 Was removed all UiComponent:::warning functions
28c4077 Wed Mar 13 11:58:39 2019 -0300 removed function urlCreator
7d09e0e Wed Mar 13 11:49:35 2019 -0300 custom buttons of Buttonactions
ed72adf Wed Mar 13 11:49:23 2019 -0300 Add Icons for button actions.
b228fba Wed Mar 13 11:49:01 2019 -0300 custom buttons of Buttonactions

web/constant.php
<?php

// comment out the following three lines when deployed to production
defined('YII_DEBUG') || define('YII_DEBUG', true);
defined('YII_TRACE_LEVEL') || define('YII_TRACE_LEVEL', 3);
defined('YII_ENV') || define('YII_ENV', 'dev'); // 'prod' is for production environment

defined('DISABLE_CACHE') || define('DISABLE_CACHE', true);


// BEGIN GLOBAL CONSTANTS of this web application --------------------------------------------------
//
defined('ACTION_CREATE') || define('ACTION_CREATE', 'create');
defined('ACTION_DELETE') || define('ACTION_DELETE', 'delete');
defined('ACTION_INDEX')  || define('ACTION_INDEX', 'index');
defined('ACTION_LOGOUT')   || define('ACTION_LOGOUT', 'logout');
defined('ACTION_REMOVE') || define('ACTION_REMOVE', 'remove');

defined('ACTIONS') || define('ACTIONS', 'actions');
defined('ACTION_UPDATE') || define('ACTION_UPDATE', 'update');
defined('ACTION_VIEW')   || define('ACTION_VIEW', 'view');
defined('ACTIVE') || define('ACTIVE', 'active');
defined('ALLOW')   || define('ALLOW', 'allow');
defined('ATTRIBUTE') || define('ATTRIBUTE', 'attribute');
defined('ATTRIBUTES') || define('ATTRIBUTES', 'attributes');
defined('AUTOCOMPLETE') || define('AUTOCOMPLETE', 'autocomplete');
defined('AUTOFOCUS') || define('AUTOFOCUS', 'autofocus');
defined('BREADCRUMBS') || define('BREADCRUMBS', 'breadcrumbs');
defined('COLSM1') || define('COLSM1', 'col-sm-1');
defined('CONTENT') || define('CONTENT', 'content');
defined('DATA_PROVIDER')   || define('DATA_PROVIDER', 'dataProvider');
defined('DETAILVIEW_CLASS') || define('DETAILVIEW_CLASS', 'table maxwidth table-striped ');
defined('DATEFORMAT')   || define('DATEFORMAT', 'dd-MM-yyyy');
defined('DATETIMEFORMAT')   || define('DATETIMEFORMAT', 'DD-MM-YYYY HH:mm');
defined('ERROR')   || define('ERROR', 'error');
defined('ERROR_MODULE') || define('ERROR_MODULE', 'Failed to {module}, error: {error}');
defined('FILTER') || define('FILTER', 'filter');
defined('FORMAT') || define('FORMAT', 'format');
defined('GRID_DATACOLUMN') || define('GRID_DATACOLUMN', 'yii\grid\DataColumn');
defined('GRIDVIEW_CSS') || define('GRIDVIEW_CSS', 'table maxwidth items table-striped table-condensed');
defined('HEADER') || define('HEADER', 'header');
defined('HTML_BTN_BTN_DEFAULT') || define('HTML_BTN_BTN_DEFAULT', 'btn btn-default');
defined('HTML_CLOSE_DIV_OPEN_COL_SM_8') || define('HTML_CLOSE_DIV_OPEN_COL_SM_8', '</div><div class="col-sm-8">');
defined('HTML_DIV_CLOSE') || define('HTML_DIV_CLOSE', '</div>');
defined('HTML_DIV_CLOSE_DIV6_OPEN') || define('HTML_DIV_CLOSE_DIV6_OPEN', '</div><div class="col-sm-6">');
defined('HTML_DIV_CLOSEX2') || define('HTML_DIV_CLOSEX2', '</div></div>');
defined('HTML_MODEL_LG') || define('HTML_MODEL_LG', 'model-lg');
defined('HTML_OPTION_CLOSE') || define('HTML_OPTION_CLOSE', '</option>');
defined('HTML_OPTION') || define('HTML_OPTION', '<option>');
defined('HTML_ROW_DIV6') || define('HTML_ROW_DIV6', '<div class="row"><div class="col-sm-6">');
defined('HTML_ROW_OPEN_COL_SM_4') || define('HTML_ROW_OPEN_COL_SM_4', '<div class="row"><div class="col-sm-4">');
defined('HTML_SPACEX2') || define('HTML_SPACEX2', '&nbsp; &nbsp;');
defined('HTML_WEBPAGE_CLOSE') || define('HTML_WEBPAGE_CLOSE', '</div></div></div>');
defined('HTML_WEBPAGE_CLOSE_OPEN_COL_SM_4') || define('HTML_WEBPAGE_CLOSE_OPEN_COL_SM_4', '</div><div class="col-sm-4">');
defined('HTML_WEBPAGE_OPEN_COL_SM_8') || define('HTML_WEBPAGE_OPEN_COL_SM_8', '<div class="webpage"><div class="row"><div class="col-sm-8 box">');
defined('HTML_WEBPAGE_OPEN')  || define('HTML_WEBPAGE_OPEN','<div class=" box">');
defined('ID') || define('ID', 'id');
defined('INFO') || define('INFO', 'info');
defined('INPUT') || define('INPUT', 'input');
defined('INPUT_OPTIONS') || define('INPUT_OPTIONS', 'inputOptions');
defined('INPUT_TEMPLATE') || define('INPUT_TEMPLATE', 'inputTemplate');
defined('ITEMS') || define('ITEMS', 'items');
defined('LABEL') || define('LABEL', 'label');
defined('LABELOPTIONS') || define('LABELOPTIONS', 'labelOptions');
defined('LANGUAGE') || define('LANGUAGE', 'label');
defined('LENGTH') || define('LENGTH', 'length');
defined('MAXLENGTH') || define('MAXLENGTH', 'maxlength');
defined('METHOD') || define('METHOD', 'method');
defined('MODEL')   || define('MODEL', 'model');
defined('MODULE') || define('MODULE', 'module');
defined('MSG_ERROR') || define('MSG_ERROR', 40);
defined('MSG_INFO') || define('MSG_INFO', 10);
defined('MSG_SECURITY_ISSUE')   || define('MSG_SECURITY_ISSUE', 50);
defined('MSG_SUCCESS') || define('MSG_SUCCESS', 20);
defined('MSG_WARNING') || define('MSG_WARNING', 30);
defined('OPTIONS') || define('OPTIONS', 'options');
defined('PAGE_SIZE') || define('PAGE_SIZE', 'pageSize');
defined('PATTERN_DATE') || define('PATTERN_DATE', 'd-m-Y');
defined('PATTERN_DATETIME') || define('PATTERN_DATETIME', 'Y-m-d H:m:i');
defined('PATTERN_DATE_YMD') || define('PATTERN_DATE_YMD', 'Y-m-d');
defined('PATTERN') || define('PATTERN', 'pattern');
defined('PATTERN_PHONE') || define('PATTERN_PHONE', '[0-9]{3}-[0-9]{4}-[0-9]{4}');
defined('PLACEHOLDER') || define('PLACEHOLDER', 'placeholder');
defined('PROMPT') || define('PROMPT', 'prompt');
defined('RANGE')   || define('RANGE', 'range');
defined('REQUIRED') || define('REQUIRED', 'required');
defined('ROLES')   || define('ROLES', 'roles');
defined('SEARCH_MODEL')   || define('SEARCH_MODEL', 'searchModel');
defined('STR_CLASS')   || define('STR_CLASS', 'class');
defined('STR_DEFAULT') || define('STR_DEFAULT', 'default');
defined('STRING')   || define('STRING', 'string');
defined('SUCCESS')   || define('SUCCESS', 'success');
defined('TABINDEX') || define('TABINDEX', 'tabindex');
defined('TEMPLATE') || define('TEMPLATE', 'template');
defined('TITLE') || define('TITLE', 'title');
defined('TRANSACTION_MODULE') || define('TRANSACTION_MODULE', '{module} {method}');
defined('TYPE') || define('TYPE', 'type');
defined('UNCHECK') || define('UNCHECK', 'uncheck');
defined('VALUE') || define('VALUE', 'value');
defined('WARNING') || define('WARNING', 'warning');
